streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -204,12 +221,11 @@ void unlock(final TaskId taskId) throws IOException {
      * @param cleanupDelayMs only remove directories if they haven't been modified for at least
      *                       this amount of time (milliseconds)
      */
-    public void cleanRemovedTasks(final long cleanupDelayMs) {
+    public synchronized void cleanRemovedTasks(final long cleanupDelayMs) {
```

Comment:
I think we should leave it. We should cleanup old state dirs, but leave recently touched dirs as they would be useful in case of failure